server/container/core/src/main/java/org/apache/james/server/core/MimeMessageModule.java

```diff
@@ -0,0 +1,54 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.server.core;
+
+import java.util.regex.Pattern;
+import javax.mail.internet.MimePart;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.AbstractModule;
+
+public class MimeMessageModule extends AbstractModule {
+
+    @Override
+    protected void configure() {
+        System.setProperty("mail.mime.contenttypehandler", ContentTypeCleaner.class.getName());
+    }
+
+    public static class ContentTypeCleaner {
+        private static final Logger LOGGER = LoggerFactory.getLogger(ContentTypeCleaner.class);
+        private static final Pattern REGEX = Pattern.compile("^[\\w\\-]+/[\\w\\-]+");
```

Comment:
I am a REGEX noob. Could you explain a bit more what this regex is doing?

Reply:
It looks for a string beginning by a bunch alphanumeric characters or '-' or '_' followed by a '/', then a bunch alphanumeric characters or '-' or '_'.